tests/conftest.py

```diff
@@ -17,6 +18,128 @@
 os.environ["VLLM_WORKER_MULTIPROC_METHOD"] = "spawn"
 
 
+def pytest_generate_tests(metafunc):
+    """This hook is called during the collection phase, 
+    specifically when Pytest encounters a test function that 
+    needs parametrization. It receives a metafunc object, 
+    which provides information about the test function and 
+    allows for dynamic parametrization."""
+
+    # default parameterizations
+    default_warmup_shape = [[(64, 20, 4)]]
+    default_max_num_seqs = [4]
+    default_max_model_len = [256]
+
+    existing_markers = [
+        marker.name if marker.name != "parametrize" else marker.args[0]
+        for marker in metafunc.definition.own_markers
+    ]
+
+    marker = metafunc.config.option.markexpr  # From CLI
+    if "full_model" in marker:
+        # When -m full_model is called, all tests tagged with
+        # full_model mark will be injected with these custom values
+        if metafunc.definition.get_closest_marker("full_model"):
+            _add_param(
+                "model",
+                ["ibm-granite/granite-3.3-8b-instruct"],
+                metafunc,
+                existing_markers,
+            )
+            _add_param(
+                "backend",
+                ["sendnn"],
+                metafunc,
+                existing_markers,
+            )
+            if "not cb" in marker:
```

Comment:
🤔 

Trying to understand why this conditional is here- don't we always want to run full models with the larger warmup shapes and max model length?

Reply:
I guess? I wasn't sure what the default for CB should be